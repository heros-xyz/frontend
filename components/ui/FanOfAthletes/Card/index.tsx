import { Box, Flex, Heading, Text, WrapItem, BoxProps } from "@chakra-ui/react";
import React from "react";
import dayjs from "dayjs";
import { Clock } from "@/components/svg/Clock";
import { IFanInfo } from "@/types/athlete/types";
import HerosImage from "@/components/common/HerosImage";
interface YourAthleteCardProps extends BoxProps {
  item?: IFanInfo;
  isAdmin?: boolean;
  dateFormat: string;
  onClickItem?: (item: IFanInfo) => void;
}

const YourAthleteCard: React.FC<YourAthleteCardProps> = ({
  item,
  dateFormat = "MMM YYYY",
  isAdmin,
  onClickItem,
  ...props
}) => {
  const joinedDate = dayjs(item?.joinedDate ?? item?.createdAt).format(
    dateFormat
  );
  return (
    <Box
      borderBottomWidth="thin"
      borderColor="grey.100"
      bg="white"
      py="4"
      cursor="pointer"
      onClick={() => onClickItem && item && onClickItem(item)}
      {...props}
    >
      <Flex alignItems="center">
        <WrapItem>
          <HerosImage
            src={item?.avatar ?? ""}
            width={{ base: "50px", lg: "80px" }}
            height={{ base: "50px", lg: "80px" }}
          />
        </WrapItem>
        <Box pl={4}>
          <Heading fontSize={{ base: "12px", lg: "18px" }} color="primary">
            {item?.nickName ?? item?.fullName}
          </Heading>
          <Text
            as="span"
            color="secondary"
            fontSize={["xs", "md"]}
            fontWeight={["medium", "normal"]}
          >
            {isAdmin ? item?.sportName : "Bronze Tier"}
          </Text>
          <Flex
            fontSize={["xs", "md"]}
            alignItems="center"
            alignSelf="center"
            pt={["2", "1"]}
          >
            <Clock
              color="primary"
              width={["15px", "18px"]}
              height={["15px", "18px"]}
            />
            <Text color="primary" fontWeight="medium" pl={["2", "2.5"]}>
              Joined Date: {joinedDate}
            </Text>
          </Flex>
        </Box>
      </Flex>
    </Box>
  );
};

export default YourAthleteCard;
