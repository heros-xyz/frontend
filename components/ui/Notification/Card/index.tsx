import React from "react";
import { Box, Flex, Heading, Image, Text } from "@chakra-ui/react";
import Link from "next/link";
import { INotificationInfo } from "@/types/notifications/types";
import {
  convertDateFromNow,
  getLinkByNotificationType,
  notificationContent,
} from "@/utils/functions";
import { useMaskNotificationMutation } from "@/api/global";
import { getImageLink } from "@/utils/link";
interface IProps {
  item?: INotificationInfo;
}

const NotificationCard: React.FC<IProps> = ({ item }) => {
  const [onMaskNotification] = useMaskNotificationMutation();

  const onClickNotification = () => {
    if (item?.id) {
      onMaskNotification(item?.id);
    }
  };
  return (
    <Link
      href={getLinkByNotificationType(item) ?? ""}
      onClick={onClickNotification}
    >
      <Box>
        <Flex
          py={[2.5, 4]}
          px={5}
          gap={2.5}
          bg={`${item?.readAt === null ? "accent.1" : "grey.0"} `}
          borderRadius={["0", "10px"]}
        >
          <Image
            src={getImageLink(item?.source?.avatar)}
            width={["50px", "60px"]}
            height={["50px", "60px"]}
            alt="avatar"
            rounded="full"
            objectFit="cover"
          />
          <Box
            fontSize={["xs", "md"]}
            color={`${item?.readAt !== null && "white"}`}
            flex={1}
          >
            <Heading as="span" color="primary" fontSize={["xs", "md"]}>
              {item?.source?.nickName ?? item?.source?.fullName}
            </Heading>
            <Text as="span" color="primary" wordBreak="break-word">
              {notificationContent(item)}
            </Text>
            <Text
              fontFamily="heading"
              fontWeight="bold"
              color={`${item?.readAt === null ? "secondary" : "accent.2"}`}
            >
              {convertDateFromNow(item?.createdAt)}
            </Text>
          </Box>
        </Flex>
      </Box>
    </Link>
  );
};

export default NotificationCard;
