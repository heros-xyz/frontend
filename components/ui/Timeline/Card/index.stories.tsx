import { Flex } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import JourneyCard from "./index";
const dataCurrent = {
  title: "Join “Jaworznickie Stowarzyszenie Piłkarskie Szczakowianka Club",
  description:
    "I became the best new player on team I became the best new player on team I became the best new player on team I became the best new player on team",
  from: "10/01/2022",
  to: "10/24/2022",
  isArchive: true,
  isCurrent: true,
};
const data = {
  title: "Join England Club",
  description: "I became the best new player on team",
  from: "10/24/2022",
  isArchive: true,
  isCurrent: false,
};
const meta: Meta<typeof JourneyCard> = {
  title: "Components/TimeLineJourney/JourneyCard",
  component: JourneyCard,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof JourneyCard>;

export const JourneyCardStory: Story = {
  render: () => (
    <Flex flexDirection="column" gap={3} p={5} bg="primary" h="100vh">
      <JourneyCard item={dataCurrent} />
      <JourneyCard item={data} />
    </Flex>
  ),
};
