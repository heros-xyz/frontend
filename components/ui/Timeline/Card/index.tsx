import { Box, Flex, Heading, Text } from "@chakra-ui/react";
import dayjs from "dayjs";
import React from "react";
import { Else, If, Then } from "react-if";
import { OutlineArrowIcon } from "@/components/svg/OutlineArrowIcon";
import { ITimeLineInfo } from "..";

interface IProps {
  item: ITimeLineInfo;
  canEdit?: boolean;
  setItemEdit?: (item: ITimeLineInfo) => void;
  isCurrent: boolean;
}

const nth = (date: string) => {
  const last = +date.slice(-2);
  if (last > 3 && last < 21) return "th";
  switch (last % 10) {
    case 1:
      return "st";
    case 2:
      return "nd";
    case 3:
      return "rd";
    default:
      return "th";
  }
};

const getDay = (date: string) => {
  return dayjs(date).format("D");
};

const JourneyCard: React.FC<IProps> = ({
  item,
  canEdit,
  setItemEdit,
  isCurrent,
}) => {
  return (
    <Flex
      p={{ base: 2.5, lg: 4 }}
      pr={0}
      w={{ base: "80%", xl: "540px" }}
      color="primary"
      justifyContent="space-between"
      alignItems="center"
      // bg={isCurrent ? "accent.3" : "accent.1"}
      bg="grey.0"
      rounded="md"
      cursor={canEdit ? "pointer" : ""}
      onClick={() => setItemEdit && setItemEdit(item)}
    >
      <Box pr={2} w={{ base: "170px", sm: "75%" }}>
        <Heading fontSize={{ base: "xs", lg: "lg" }} mb={1}>
          {item.title}
        </Heading>
        <Text fontSize={{ base: "xxs", lg: "md" }}>{item.description}</Text>
      </Box>
      <Flex w={{ lg: "145px" }}>
        <Box w="1px" h={{ base: "51px", xl: "74px" }} bg="grey.300" />
        <If condition={item?.endDate}>
          <Then>
            <Box
              textAlign="center"
              fontSize="xxs"
              w={20}
              fontWeight="medium"
              flex={1}
            >
              <Box fontSize={{ base: "xxs", lg: "md" }}>
                <Text as="span">{getDay(item?.startDate)}</Text>
                <Text as="sup" fontSize={{ base: "8px", lg: "xxs" }}>
                  {nth(getDay(item?.startDate))}
                </Text>
                <Text as="span">
                  {dayjs(item?.startDate).format(" MMM YYYY")}
                </Text>
              </Box>
              <OutlineArrowIcon color="primary" />
              <Box fontSize={{ base: "xxs", lg: "md" }}>
                <Text as="span">{getDay(item?.endDate || "")}</Text>
                <Text as="sup" fontSize={{ base: "8px", lg: "xxs" }}>
                  {nth(getDay(item?.endDate || ""))}
                </Text>
                <Text as="span">
                  {dayjs(item?.endDate).format(" MMM YYYY")}
                </Text>
              </Box>
            </Box>
          </Then>
          <Else>
            <Box
              textAlign="center"
              pt={1}
              w={20}
              flex={1}
              fontSize={{ base: "xs", lg: "md" }}
            >
              <Box mb={1} fontWeight={500}>
                <Text as="span">{getDay(item?.startDate)}</Text>
                <Text as="sup" fontSize={{ base: "8px", lg: "xxs" }}>
                  {nth(getDay(item?.startDate))}
                </Text>
                <Text as="span">{dayjs(item?.startDate).format(" MMM")}</Text>
              </Box>
              <Text
                as="b"
                fontFamily="heading"
                fontSize={{ base: "md", xl: "xl" }}
              >
                {dayjs(item?.startDate).format("YYYY")}
              </Text>
            </Box>
          </Else>
        </If>
      </Flex>
    </Flex>
  );
};

export default JourneyCard;
