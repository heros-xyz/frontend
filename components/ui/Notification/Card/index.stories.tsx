import { Box } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import Card from "./index";

const item = {
  id: "1",
  type: "A_NEW_INTERACTION",
  message: "notification.athlete-new-interaction",
  source: {
    id: "48",
    avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
    fullName: "Sam Smith",
  },
  interaction: {
    id: "70",
    content: "some content should be trimmed if too long",
  },
  comment: {
    id: "b4e25260-1716-4c49-9e31-901551718c60",
    content: "eight",
    userId: "b59d63f8-8fdb-4f23-a9f1-dfe4572594af",
    createdAt: "2023-02-23T07:52:51.304Z",
  },
  readAt: null,
  createdAt: "2023-02-22T08:43:57.555Z",
};

const meta: Meta<typeof Card> = {
  title: "Components/Card",
  component: Card,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof Card>;

export const CardNotificationStory: Story = {
  render: (args) => (
    <Box bg="primary">
      <Card {...args} item={item} />
    </Box>
  ),
};
