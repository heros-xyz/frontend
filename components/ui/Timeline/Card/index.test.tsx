import { Box } from "@chakra-ui/react";
import { render } from "@testing-library/react";
import Card from "./index";

const dataCurrent = {
  title: "Join England Club",
  description: "I became the best new player on team",
  from: "10/01/2022",
  to: "10/24/2022",
  isArchive: true,
  isCurrent: true,
};
const data = {
  title: "Join England Club",
  description: "I became the best new player on team",
  from: "10/24/2022",
  isArchive: true,
  isCurrent: false,
};

it("renders homepage unchanged", () => {
  const { container } = render(
    <Box>
      <Card item={data} />
      <Card item={dataCurrent} />
    </Box>
  );
  expect(container).toMatchSnapshot();
});
