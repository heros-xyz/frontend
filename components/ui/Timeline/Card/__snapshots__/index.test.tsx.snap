// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-q9og19"
    >
      <div
        class="css-1jaf54"
      >
        <h2
          class="chakra-heading css-109tgpy"
        >
          Join “Jaworznickie Stowarzyszenie Piłkarskie Szczakowianka Club
        </h2>
        <p
          class="chakra-text css-s2a0zi"
        >
          I became the best new player on team I became the best new player on team I became the best new player on team I became the best new player on team
        </p>
      </div>
      <div
        class="css-64rv73"
      >
        <div
          class="css-xt3441"
        />
        <div
          class="css-1kmzqxq"
        >
          <div
            class="css-s2a0zi"
          >
            <span
              class="chakra-text css-0"
            >
              1
            </span>
            <sup
              class="chakra-text css-cky0vs"
            >
              st
            </sup>
            <span
              class="chakra-text css-0"
            >
               Oct 2022
            </span>
          </div>
          <div
            class="css-1vp8jwk"
          >
            <div
              class="css-1nz5tvn"
            />
            <div
              class="css-4qxy49"
            >
              <svg
                class="chakra-icon css-q21vef"
                focusable="false"
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 7.5L7 9.5M7 9.5L9 7.5M7 9.5L7 4.5M1 7C1 3.68629 3.68629 0.999999 7 0.999999C10.3137 1 13 3.68629 13 7C13 10.3137 10.3137 13 7 13C3.68629 13 1 10.3137 1 7Z"
                  stroke="#313F4C"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="0.5"
                />
              </svg>
            </div>
            <div
              class="css-159arjl"
            />
          </div>
          <div
            class="css-s2a0zi"
          >
            <span
              class="chakra-text css-0"
            >
              24
            </span>
            <sup
              class="chakra-text css-cky0vs"
            >
              th
            </sup>
            <span
              class="chakra-text css-0"
            >
               Oct 2022
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-q9og19"
    >
      <div
        class="css-1jaf54"
      >
        <h2
          class="chakra-heading css-109tgpy"
        >
          Join “Jaworznickie Stowarzyszenie Piłkarskie Szczakowianka Club
        </h2>
        <p
          class="chakra-text css-s2a0zi"
        >
          I became the best new player on team I became the best new player on team I became the best new player on team I became the best new player on team
        </p>
      </div>
      <div
        class="css-64rv73"
      >
        <div
          class="css-174s0jl"
        />
        <div
          class="css-1kmzqxq"
        >
          <div
            class="css-s2a0zi"
          >
            <span
              class="chakra-text css-0"
            >
              1
            </span>
            <sup
              class="chakra-text css-cky0vs"
            >
              st
            </sup>
            <span
              class="chakra-text css-0"
            >
               Oct 2022
            </span>
          </div>
          <div
            class="css-1vp8jwk"
          >
            <div
              class="css-1nz5tvn"
            />
            <div
              class="css-4qxy49"
            >
              <svg
                class="chakra-icon css-q21vef"
                focusable="false"
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 7.5L7 9.5M7 9.5L9 7.5M7 9.5L7 4.5M1 7C1 3.68629 3.68629 0.999999 7 0.999999C10.3137 1 13 3.68629 13 7C13 10.3137 10.3137 13 7 13C3.68629 13 1 10.3137 1 7Z"
                  stroke="#313F4C"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="0.5"
                />
              </svg>
            </div>
            <div
              class="css-159arjl"
            />
          </div>
          <div
            class="css-s2a0zi"
          >
            <span
              class="chakra-text css-0"
            >
              24
            </span>
            <sup
              class="chakra-text css-cky0vs"
            >
              th
            </sup>
            <span
              class="chakra-text css-0"
            >
               Oct 2022
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
